src/internal/proc.js

```diff
@@ -420,6 +408,28 @@ export default function proc(
     cb.cancel = takeCb.cancel
   }
 
+  function runDeclarativeEffect(effect, effectId, cb) {
+    if (options.middleware) {
```

Comment:
Oh wait nvm I think I get the point. In case the effect gets swaped by something thats not declarative.

Will take care of this